components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/CertificateValidationUtil.java

```diff
@@ -1223,4 +1320,155 @@ private static boolean isSelfSignedCert(X509Certificate cert) {
         }
     }
 
+    private static void addDefaultValidatorConfigToConfigStore(OMElement validatorsElement, String tenantDomain)
+            throws CertificateValidationException {
+
+        List<Validator> defaultValidatorConfig = getDefaultValidatorConfig(validatorsElement);
+        for (Validator validator : defaultValidatorConfig) {
+            log.debug("Adding the configurations for validator: " + validator.getDisplayName() + " to the config " +
+                            "store for tenant: " + tenantDomain);
+            org.wso2.carbon.identity.configuration.mgt.core.model.Resource newResource =
+                    buildResourceFromValidator(validator, getNormalizedName(validator.getDisplayName()),
+                            VALIDATOR_RESOURCE_TYPE);
+            addResource(newResource);
+        }
+    }
+
+    /**
+     * Method to add a resource.
+     *
+     * @param newResource   New resource to be added.
+     */
+    private static void addResource(
+            org.wso2.carbon.identity.configuration.mgt.core.model.Resource newResource) {
+
+        try {
+            CertValidationDataHolder.getInstance().getConfigurationManager()
+                    .addResource(newResource.getResourceType(), newResource);
+        } catch (ConfigurationManagementException e) {
+            if (ERROR_CODE_RESOURCE_ALREADY_EXISTS.getCode().equals(e.getErrorCode())) {
+                log.debug("Resource already exists in the tenant config store.");
+            } else if (ERROR_CODE_RESOURCE_TYPE_DOES_NOT_EXISTS.getCode().equals(e.getErrorCode())) {
+                log.error("Resource type does not exist in the tenant config store.");
+            } else {
+                log.error("Error while adding validator configurations.", e);
+            }
```

Comment:
Shouldn't it throw errors in these cases?

Reply:
removed the part